docker/dyno/app/api/docker.py

```diff
@@ -0,0 +1,228 @@
+    config = low_client.inspect_container(_normalize_name(container))['HostConfig']
+    """
+    cpu_quota (int) - Limit CPU CFS (Completely Fair Scheduler) quota
+        -> HostConfig.CpuShares
+    blkio_weight (int) - Block IO (relative weight), between 10 and 1000
+        -> HostConfig.BlkioWeight
+    mem_limit (str) - Memory limit with units, such as 4m. The following
+    are supported: b, k, m, g
+        -> HostConfig.Memory
+    """
+    ret = {
+        'CPU': _denormalize_value('cpu', config['CpuQuota']),
+        'Mem': _denormalize_value('mem', config['Memory']),
+        # 'IO': _denormalize_value('io', config['BlkioWeight']),
+    }
+    return ret
+
+@bp.route('/update', methods=['GET'])
+def update():
+    """
+    Update a container setting
+
+    We take the following required args:
+    c: (str) The container name
+
+    """
+    c = _normalize_name(request.args.get('c'))
+    component = request.args.get('component')
+    val = int(request.args.get('val'))
+    config = {
+        'container': c,
+        'settings': {}
+    }
+    c = component.lower()
+    if c == 'cpu':
+        config['settings']['cpu_quota'] = _normalize_value(c, val)
+    if c == 'io':
+        config['settings']['blkio_weight'] = _normalize_value(c, val)
+    if c == 'mem':
+        config['settings']['mem_limit'] = str(_normalize_value(c, val)) + "m"
+        config['settings']['memswap_limit'] = -1
+    #print(config['settings'])
+    c = client.containers.get(config['container'])
+    c.update(**config['settings'])
+    return {}
+
+def _range():
+    range_path = os.path.join(app.app.root_path, 'range.yml')
+    with open(range_path, 'r') as fh_:
+        slider_range = yaml.load(fh_, Loader=yaml.FullLoader)
+    return slider_range
+
+def _denormalize_value(code, val):
+    """
+    Take a current value and return the percentage val
+    """
+    slider_range = _range()
+    lval, uval = slider_range[code]
+    ret = ((val - min([uval, lval])) / (max([lval, uval]) - min([lval-uval]))) + 1
+    return int(ret)
+    # val_range = abs(uval - lval)
+    # if lval < uval:
+    #     return int(100 - ((val / val_range) * 100))
+    # else:
+    #     return int((val / val_range) * 100)
+
+
+def _normalize_value(code, val):
+    """
+    This uses the range.yml configuration file which populates
+    a set of values to determine the upper and lower range.
+    We take our input value from the web interface to this function
+    which is in the range of 0-100 and we turn that into an actual
+    value to pass to the toxic
+    """
+    slider_range = _range()
+
+    lval, uval = slider_range[code]
+
+    # val_range = abs(uval - lval) + 1
+    # # if lval < uval:
+    # #     ret = abs(uval - int(val_range * (val / 100)))
+    # # else:
+    # #     # ret = int(val_range * (val / 100))
+    # #     adder = range / val 
+    # #     ret = uval + adder
+    #print('received: ', val)
+    #print('range vals', lval, uval)
```

Comment:
NIT:

If not required, can it be removed?

Reply:
Yup! Done. Thanks!